src/models/callercalleeproxy.h

```diff
@@ -44,5 +40,16 @@ class CallerCalleeProxy : public QSortFilterProxyModel
         return CallerCalleeProxyDetail::match(this, key);
     }
 
+    bool lessThan(const QModelIndex& source_left, const QModelIndex& source_right) const override
```

Comment:
de-inline

Reply:
sadly this is not possible, since Qt does not offer a compare function for QVariant (they added one in Qt 6.2)

Reply:
I'm sorry, but you are doing it wrong™ 

the source model does not matter here, the only thing that matters is the role which should be Model::SortRole and could be hard coded in the ctor here already.

as such, we can expect (and assert!) that the values we get here are actually locations for the location column! Meaning, rewrite this as:

```
Q_ASSERT(sortRole() == Model::SortRole);
if (source_left.column() == source_right.column() && source_left.column() == Model::Location)
{
    return ...Detail::lessThanLocation(source_left, source_right, Model::SortRole);
}
```

Reply:
The source model matters, `ResultsCallerCalleePage` create a `CallerCalleeProxy` with a `CallerModel`, a `CalleeModel` and a `SourceMapModel`. Only the `SourceMapModel` has the `Location` column.

I should create a new proxy only for the `SourceMapModel` to handle this.

Reply:
or use a `std::optional` or similar to pass the location column, and only when that's given use this feature.